### PR TITLE
Add-webhook-support

### PR DESCRIPTION
**Pull request by zkldi**
Fixes #19

This is a fairly huge patch, as it also adds conf.json5 support, and a lot of utility functions for interacting with the tachi API.

Review is massively appreciated.

### CODE REVIEW

**Comment by zkldi** [comment]
important: https://github.com/TeamNewGuys/tachi-bot/pull/20/commits/3e647a69b038f86e2ba34e5748343ce6c5f72f16#diff-64dd49d8be0fae3fb7e83af115350471cf5a79c0e94059cffd03d8a0b5c73e77L34

This code will no longer build - `strict` mode in TS throws an error here. Is there meant to be a null check, or should this just be asserted out?

File: src/setup.ts [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+
+
+/**
+ * The bot config is parsed from a JSON5 file, and supports complex
+ * data structures like objects/arrays.
+ * 
+ * This shouldn't store any secrets, and should instead store various
+ * configuration variables.
+ */
+export interface BotConfig {
```
Comment (Puffycheeses) [inline src/setup.ts:25]: Comments inside these interfaces are not necessary, the variable names are relatively self-explanatory.

File: src/setup.ts [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+		throw err;
+	}
+
+	return data;
+}
+
+/**
+ * In contrast to BotConfig, this should contain important environment secrets.
+ */
+export interface ProcessEnvironment {
```
Comment (Puffycheeses) [inline src/setup.ts:70]: As above, these variable names are relatively self-explanatory.

File: src/setup.ts [hunk 3]
```diff
@@ -0,0 +1,109 @@
+import { LoggerLayers } from "config";
+import { createLayeredLogger } from "utils/logger";
+import { FormatPrError } from "utils/prudence";
+import JSON5 from "json5";
+import fs from "fs";
+import p from "prudence";
```
Comment (Puffycheeses) [inline src/setup.ts:6]: Not too sure what prudence is (will be googling 👍) but I think using:
```ts
import prudence from "prudence";
```
will make some of the code below more readable.

```ts
const err = prudence(data, {
	SERVER_PORT: p.isPositiveNonZeroInteger,
	TACHI_SERVER_LOCATION: "string",
	OUR_URL: "string",
});
```

**Review by Puffycheeses — CHANGES_REQUESTED** [review]

**Review by Puffycheeses — APPROVED** [review]

**Pull request merged**